nifi-nar-bundles/nifi-questdb-bundle/nifi-questdb/src/main/java/org/apache/nifi/questdb/embedded/EmbeddedDatabaseManager.java

```diff
@@ -0,0 +1,294 @@
+                    }
+                } else if (!databaseFiles.get(tableDefinition.getName()).isDirectory()) {
+                    throw new CorruptedDatabaseException(String.format("Table %s cannot be created because there is already a file exists with the given name", tableDefinition.getName()));
+                }
+            }
+
+            // Checking if tables are healthy.
+            for (final ManagedTableDefinition tableDefinition : context.getTableDefinitions()) {
+                try {
+                    final TableToken tableToken = this.engine.get().getTableTokenIfExists(tableDefinition.getName());
+                    final TableRecordMetadata metadata = this.engine.get().getSequencerMetadata(tableToken);
+                    metadata.close();
+
+                    client.execute(String.format("SELECT * FROM %S LIMIT 1", tableDefinition.getName()));
+                } catch (final Exception e) {
+                    throw new CorruptedDatabaseException(e);
+                }
+            }
+        } finally {
+            try {
+                client.disconnect();
+            } catch (DatabaseException e) {
+                throw new CorruptedDatabaseException(e);
+            }
+        }
+    }
+
+    private void startRollover() {
+        final RolloverWorker rolloverWorker = new RolloverWorker(acquireClient(), context.getTableDefinitions());
+        final ScheduledFuture<?> rolloverFuture = scheduledExecutorService.scheduleWithFixedDelay(
+                rolloverWorker, context.getRolloverFrequency().toMillis(), context.getRolloverFrequency().toMillis(), TimeUnit.MILLISECONDS);
+        scheduledFutures.add(rolloverFuture);
+        LOGGER.debug("Rollover started");
+    }
+
+    private void stopRollover() {
+        LOGGER.debug("Rollover shutdown started");
+
+        int cancelCompleted = 0;
+        int cancelFailed = 0;
+        for (final ScheduledFuture<?> scheduledFuture : scheduledFutures) {
+            final boolean cancelled = scheduledFuture.cancel(true);
+            if (cancelled) {
+                cancelCompleted++;
+            } else {
+                cancelFailed++;
+            }
+        }
+
+        LOGGER.debug("Rollover shutdown task cancellation status: completed [{}] failed [{}]", cancelCompleted, cancelFailed);
+        final List<Runnable> tasks = scheduledExecutorService.shutdownNow();
+        LOGGER.debug("Rollover Scheduled Task Service shutdown remaining tasks [{}]", tasks.size());
+    }
+
+    private Client getUnmanagedClient() {
+        return new EmbeddedClient(() -> engine.get());
+    }
+
+    public Client acquireClient() {
+        checkIfManagerIsInitialised();
+        final Client fallback = new NoOpClient();
+
+        if (state.get() == EmbeddedDatabaseManagerStatus.CORRUPTED) {
+            LOGGER.warn("The database is corrupted. Dummy client is returned");
```

Comment:
I recommend changing this to and error and adjusting the wording to indicate the behavior.
```suggestion
            LOGGER.error("The database is corrupted: Status History will not be stored");
```